### ISSUE TITLE
Lint/ShadowingOuterLocalVariable false positive

### ISSUE THREAD

**Issue by davide125** [bbatsov/rubocop#1987]
This minimal example
<pre>
foo = true
bar = {}
if foo
  fs = 'cake'
  puts fs
else
  bar.each do |device, fs|
    puts device if fs == 'baz'
  end
end
</pre>
fails with
<pre>
/tmp/test.rb:7:24: W: Shadowing outer local variable - fs.
  bar.each do |device, fs|
</pre>
despite the fact that fs is not really being shadowed, as it's being used in the other if branch. This snipped passes ruby -w with no issues.

**Comment by jonas054** [comment]
The `fs` variable *is* shadowed by the `fs` block parameter. If you change the value of `foo`to `false` and insert `p fs2` right after the `else` line you get
```undefined local variable or method fs2 for main:Object (NameError)```
from ruby, while `p fs` gives you `nil`, which means that `fs` is in scope.

I get `warning: shadowing outer local variable - fs` from all my installed ruby versions when running with `-w`, except `rbx-2.5.2`.

**Comment by bbatsov** [comment]
Yep, that's not a bug.
